lib/index.js

```diff
@@ -319,3 +333,134 @@ exports.socket = createSocket;
 
 exports.createDevice = createDevice;
 exports.device = createDevice;
+
+/* convenience methods to set and to get sockopts, a.k.a. the sockopt API */
+var sol = {
+  linger          : nn.NN_LINGER,
+  sndbuf          : nn.NN_SNDBUF,
+  rcvbuf          : nn.NN_RCVBUF,
+  sndtimeo        : nn.NN_SNDTIMEO,
+  rcvtimeo        : nn.NN_RCVTIMEO,
+  reconn          : nn.NN_RECONNECT_IVL,
+  maxreconn       : nn.NN_RECONNECT_IVL_MAX,
+  sndprio         : nn.NN_SNDPRIO,
+  tcpnodelay      : nn.NN_TCP_NODELAY
+}
+
+function linger(number){
+  if(number){
```

Comment:
Should we do any validation on the inputs?  What happens if you pass all of the functions strings?